src/wp-includes/class-wp-scripts.php

```diff
@@ -714,6 +900,208 @@ public function in_default_dir( $src ) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Gets all dependents of a script.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle The script handle.
+	 * @return string[] Script handles.
+	 */
+	private function get_dependents( $handle ) {
+		// Check if dependents map for the handle in question is present. If so, use it.
+		if ( array_key_exists( $handle, $this->dependents_map ) ) {
+			return $this->dependents_map[ $handle ];
+		}
+
+		$dependents = array();
+
+		// Iterate over all registered scripts, finding dependents of the script passed to this method.
+		foreach ( $this->registered as $registered_handle => $args ) {
+			if ( in_array( $handle, $args->deps, true ) ) {
+				$dependents[] = $registered_handle;
+			}
+		}
+
+		// Add the handles dependents to the map to ease future lookups.
+		$this->dependents_map[ $handle ] = $dependents;
+
+		return $dependents;
+	}
+
+	/**
+	 * Checks if the strategy passed is a valid delayed (non-blocking) strategy.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $strategy The strategy to check.
+	 * @return bool True if $strategy is one of the delayed strategies, otherwise false.
+	 */
+	private function is_delayed_strategy( $strategy ) {
+		return in_array(
+			$strategy,
+			$this->delayed_strategies,
+			true
+		);
+	}
+
+	/**
+	 * Checks if all of a script's dependents are delayed, which is required to maintain execution order.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string              $handle     The script handle.
+	 * @param bool                $async_only Whether to limit to async strategy only.
+	 * @param array<string, true> $checked    Optional. An array of already checked script handles, used to avoid recursive loops.
+	 * @return bool True if all dependents are delayed, false otherwise.
+	 */
+	private function has_only_delayed_dependents( $handle, $async_only = false, $checked = array() ) {
+		// If this node was already checked, this script can be delayed and the branch ends.
+		if ( array_key_exists( $handle, $checked ) ) {
+			return true;
+		}
+
+		$checked[ $handle ] = true;
+		$dependents         = $this->get_dependents( $handle );
+
+		// If there are no dependents remaining to consider, the script can be deferred.
+		if ( empty( $dependents ) ) {
+			return true;
+		}
+
+		// Consider each dependent and check if it is delayed.
+		foreach ( $dependents as $dependent ) {
+			// If the dependent script has no src (as it represents a script bundle), ignore it for consideration.
+			if ( empty( $this->registered[ $dependent ]->src ) ) {
+				continue;
+			}
+
+			// If the dependency is not enqueued, ignore it for consideration.
+			if ( ! $this->query( $dependent, 'enqueued' ) ) {
+				continue;
+			}
+
+			// If the dependent script is not using the defer or async strategy, no script in the chain is delayed.
+			$strategy = $this->get_data( $dependent, 'strategy' );
+			if ( $async_only ) {
+				if ( 'async' !== $strategy ) {
+					return false;
+				}
```

Comment:
```suggestion
			if ( $async_only && 'async' !== $strategy ) {
			     return false;
```

Does this work here?

Reply:
No, but it could rather be:

```php
			if ( $async_only && 'async' !== $strategy ) {
				return false;
			} elseif ( ! $async_only && ! $this->is_delayed_strategy( $strategy ) ) {
				return false;
			}
```

Reply:
Or perhaps instead:

```php
if ( $async_only ? 
	'async' !== $strategy : 
	! $this->is_delayed_strategy( $strategy ) 
) {
	return false;
}
```

Reply:
Done in https://github.com/10up/wordpress-develop/pull/80/commits/c28fa9aa4cc13a7b1d376ba5e8a7b62c90d2d814 as part of https://github.com/10up/wordpress-develop/pull/80